src/rust/ide/view/graph-editor/src/builtin/visualization/java_script/geoMap.js

```diff
@@ -249,128 +263,135 @@ class GeoMapVisualization extends Visualization {
-            const computed = this.calculateExtremesAndPushPoints(
-                parsedData,
-                preparedDataPoints,
-                accentColor
-            )
-            latitude = computed.latitude
-            longitude = computed.longitude
-        } else {
-            if (
-                parsedData.type === SCATTERPLOT_LAYER &&
-                parsedData.data.length
-            ) {
-                const computed = this.calculateExtremesAndPushPoints(
-                    parsedData.data,
-                    preparedDataPoints,
-                    accentColor
-                )
-                latitude = computed.latitude
-                longitude = computed.longitude
-            } else if (parsedData.type === GEO_MAP && ok(parsedData.layers)) {
-                parsedData.layers.forEach((layer) => {
-                    if (layer.type === SCATTERPLOT_LAYER) {
-                        let dataPoints = layer.data || []
-                        const computed = this.calculateExtremesAndPushPoints(
-                            dataPoints,
-                            preparedDataPoints,
-                            accentColor
-                        )
-                        latitude = computed.latitude
-                        longitude = computed.longitude
-                    } else {
-                        console.warn(
-                            'Geo_Map: Currently unsupported deck.gl layer.'
-                        )
-                    }
-                })
-            }
-        }
-        return { latitude, longitude }
+    centerPoint() {
+        const { x, y } = calculateCenterPoint(this.dataPoints)
+        return { latitude: x, longitude: y }
     }
 
     /**
-     * Helper for prepareDataPoints, pushes `GEO_POINT`'s to the list, and calculates central point.
-     * @returns {{latitude: number, longitude: number}} - center.
+     * Sets size of the visualization.
+     * @param size - new size, list of two numbers containing width and height respectively.
      */
-    calculateExtremesAndPushPoints(
-        dataPoints,
-        preparedDataPoints,
-        accentColor
-    ) {
-        let latitudes = []
-        let longitudes = []
-        dataPoints.forEach((e) => {
-            if (e.type === GEO_POINT) {
-                this.pushGeoPoint(preparedDataPoints, e, accentColor)
-                latitudes.push(e.latitude)
-                longitudes.push(e.longitude)
+    setSize(size) {
+        this.dom.setAttributeNS(null, 'width', size[0])
+        this.dom.setAttributeNS(null, 'height', size[1])
+        this.mapElem.setAttributeNS(
+            null,
+            'style',
+            'width:' + size[0] + 'px;height: ' + size[1] + 'px;'
+        )
+    }
+}
+
+function prepareArray(parsedData, preparedDataPoints, accentColor) {
+    pushGeoPoints(parsedData, preparedDataPoints, accentColor)
+}
+
+function prepareGeoPoints(parsedData, preparedDataPoints, accentColor) {
```

Comment:
This function name is not descriptive enoguh for me. Coul you make it better or at least add docs to it, please?